dg/lib/fyre/openshift.py

```diff
@@ -12,9 +12,184 @@
+    use_host_path_storage
+        flag indicating whether hostpath storage shall be used
+    oc_login_command_for_remote_host
+        oc login command for logging in to OpenShift on the remote host
+    """
+
+    async with dg.utils.ssh.RemoteClient(infrastructure_node_hostname) as remoteClient:
+        await remoteClient.connect()
+        await remoteClient.execute(oc_login_command_for_remote_host)
+        await remoteClient.execute(
+            "oc " + " ".join(_get_oc_adm_taint_node_command(node, db2_edition))
+        )
+
+        await remoteClient.execute(
+            "oc " + " ".join(_get_oc_label_node_command(node, db2_edition))
+        )
+
+        await remoteClient.execute(
+            "ssh " + " ".join(_get_ssh_setsebool_container_manage_cgroup_command(node))
+        )
+
+        if use_host_path_storage:
+            await label_storage_path_from_remote_host(remoteClient, node)
+
+
+def label_storage_path(node: str):
+    """Labels the storage path on the given node
+
+    See https://www.ibm.com/support/producthub/icpdata/docs/content/SSQNUZ_latest/svc-db2/hostpath-selinux-aese.html
+
+    Parameters
+    ----------
+    node
+        node on which the storage path shall be labeled
+    """
+
+    dg.utils.process.execute_command(pathlib.Path("ssh"), _get_ssh_mkdir_command(node))
+    dg.utils.process.execute_command(pathlib.Path("ssh"), _get_ssh_chmod_command(node))
+    dg.utils.process.execute_command(
+        pathlib.Path("ssh"), _get_ssh_semanage_command(node)
+    )
+
+    dg.utils.process.execute_command(
+        pathlib.Path("ssh"), _get_ssh_restorecon_command(node)
+    )
+
+
+async def label_storage_path_from_remote_host(
+    remoteClient: dg.utils.ssh.RemoteClient, node: str
+):
+    """Labels the storage path on the given node from a remote host
+
+    See https://www.ibm.com/support/producthub/icpdata/docs/content/SSQNUZ_latest/svc-db2/hostpath-selinux-aese.html
+
+    Parameters
+    ----------
+    remoteClient
+        SSH client
+    node
+        node on which the storage path shall be labeled
+    """
+
+    await remoteClient.execute("ssh " + _join_args(_get_ssh_mkdir_command(node)))
+    await remoteClient.execute("ssh " + _join_args(_get_ssh_chmod_command(node)))
+    await remoteClient.execute("ssh " + _join_args(_get_ssh_semanage_command(node)))
+    await remoteClient.execute("ssh " + _join_args(_get_ssh_restorecon_command(node)))
+
+
+def _get_oc_adm_taint_node_command(node: str, db2_edition: str) -> list[str]:
+    return [
+        "adm",
+        "taint",
+        "node",
+        node,
+        f"icp4data=database-{db2_edition}:NoSchedule",
+    ]
+
+
+def _get_oc_label_node_command(node: str, db2_edition: str) -> list[str]:
+    return ["label", "node", node, f"icp4data=database-{db2_edition}"]
+
+
+def _get_ssh_mkdir_command(node: str) -> list[str]:
+    return [f"core@{node}", "mkdir", "--parents", "/var/home/core/data"]
+
+
+def _get_ssh_chmod_command(node: str) -> list[str]:
```

Comment:
Method name sounds generic but is having a very specific use-case. Please rename the method or add additional parameters (mode, path).